src/lib/form/Form.tsx

```diff
@@ -8,13 +8,21 @@ import { InputConfig } from '../inputs/InputBase';
 import './form.css';
 import { Button } from '../buttons/Button';
 import { ButtonPrimary } from '../buttons/ButtonPrimary';
+import ReactModal from 'react-modal';
 
 export interface FormConfig extends InputConfig {}
 
+export interface UnsavedModalProps extends Partial<ReactModal.Props> {
+  modalTitle?: string;
+  modalContent?: string;
+  modalPrimaryButtonText?: string;
+  modalCloseButtonText?: string;
+}
```

Comment:
At this point, I think it makes sense to go ahead and extract the modal to its own component (`UnsavedChangesModal`) and then move this type over to the component file.

Reply:
We could potentially keep the component nested under the `src/lib/form` directory.

Reply:
This is totally just a thought. We don't have to do it right now.